tests/unit/agents/common/test_agent.py

```diff
@@ -190,6 +190,235 @@ async def test_invoke_chain(
+                        ToolMessage(
+                            content='{"data": "response 1"}',
+                            name="query_tool",
+                            tool_call_id="tool_call_id_1",
+                        ),
+                        ToolMessage(
+                            content='{"data": "response 2"}',
+                            name="query_tool",
+                            tool_call_id="tool_call_id_2",
+                        ),
+                    ],
+                    my_task=SubTask(
+                        description="test task",
+                        task_title="test",
+                        assigned_to="KubernetesAgent",
+                    ),
+                ),
+                [{"data": "response 2"}, {"data": "response 1"}],
+                6000,
+                "Summarized tool response",
+                "Summarized",
+            ),
+            (
+                "Tool message with list content, should handle correctly",
+                BaseAgentState(
+                    is_last_step=False,
+                    messages=[],
+                    remaining_steps=25,
+                    agent_messages=[
+                        AIMessage(content="Hello"),
+                        ToolMessage(
+                            content='[{"item1": "value1"}, {"item2": "value2"}]',
+                            name="query_tool",
+                            tool_call_id="tool_call_id_1",
+                        ),
+                    ],
+                    my_task=SubTask(
+                        description="test task",
+                        task_title="test",
+                        assigned_to="KubernetesAgent",
+                    ),
+                ),
+                [{"item1": "value1"}, {"item2": "value2"}],
+                7000,
+                "Summarized tool response",
+                "Summarized",
+            ),
+        ],
+    )
+    async def test_summarize_tool_response(
+        self,
+        test_case: str,
+        given_state: BaseAgentState,
+        tool_response_objects: list,
+        token_count: int,
+        expected_result: str,
+        expected_message_content: str,
+        monkeypatch,
+    ):
+        agent = mock_agent()
+        agent.tool_response_summarization = Mock()
+        agent.tool_response_summarization.summarize_tool_response = AsyncMock(
+            return_value="Summarized tool response"
+        )
+
+        monkeypatch.setattr(
+            "agents.common.agent.compute_string_token_count",
+            lambda content, model_type: token_count,
+        )
+
+        token_limit = 1000
+        monkeypatch.setattr(
+            "agents.common.agent.TOOL_RESPONSE_TOKEN_COUNT_LIMIT",
+            {agent.model.name: token_limit},
+        )
+        monkeypatch.setattr("agents.common.agent.TOTAL_CHUNKS_LIMIT", 10)
+
+        result = await agent._summarize_tool_response(given_state, {})
```

Comment:
let's move` result = await agent._summarize_tool_response(given_state, {})` to tool function  and leave model node unchanged. I think this is the best approach.